src/main/java/seedu/recipe/storage/JsonAdaptedRecipe.java

```diff
@@ -82,41 +69,69 @@ public Recipe toModelType() throws IllegalValueException {
+        List<Ingredient> OutputIngredientList = new ArrayList<>();
+        String[] ingredientsArray= ingredients.split(",");
+        for (String ingredient : ingredientsArray) {
+            parameters = ingredient.split("[\\s+,/]"); 
+            double ingredientAmount = Double.parseDouble(parameters[0]);
+            IngredientAmountUnit ingredientUnit = new IngredientAmountUnit(parameters[1]);        
+            String ingredientName = parameters[2];
+            try{
+                HashSet<Ingredient> substitutions = new HashSet<>();
+                String[] substitutionsArray = parameters[3].split("[\\s+]");
+                for (String substitution : substitutionsArray) {
+                    Ingredient newSubstitution = new Ingredient(substitution);
+                    substitutions.add(newSubstitution);
+                }
+            }catch(ArrayIndexOutOfBoundsException e) {
+                System.out.println(e.getMessage());
+            }finally {
+                RecipeIngredient newIngredient = new RecipeIngredient(ingredientName, ingredientAmount);
+                OutputIngredientList.add(newIngredient);
+            }       
+        }    
+        /*Right now RecipeIngredient has no way to take in Unit and Substitution in constructor method,
+        so either we add it or we just set() unit and substitution under modelIngredients here.
+        */
+        Optional<List<Ingredient>> ingredients = Optional.of(OutputIngredientList);
+
+        List<Step> stepList = new ArrayList<>();
+        String[] stepArray = steps.split(",");
+        for (String step : stepArray) {
+            Step newStep = new Step(step);
+            stepList.add(newStep);
         }
-        if (!Address.isValidAddress(address)) {
-            throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
+        if (!stepList.isEmpty()) {
+            Optional<List<Step>> steps = Optional.of(stepList);
         }
-        final Address modelAddress = new Address(address);
-
-        if (step == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Step.class.getSimpleName()));
+        
+        Set<Tag> output = new HashSet<>();
+        String[] tagList = tags.split(",");
+        for (String tag : tagList) {
+            Tag newTag = new Tag(tag);
+            output.add(newTag);
         }
-        if (!Step.isValidStep(step)) {
-            throw new IllegalValueException(Step.MESSAGE_CONSTRAINTS);
+        if (!output.isEmpty()) {
+            final Optional<Set<Tag>> optionalModelTags = Optional.of(output);
         }
-        final Step modelStep = new Step(step);
-
-        final Set<Tag> modelTags = new HashSet<>(recipeTags);
-        return new Recipe(modelName, modelIngredient, modelEmail, modelAddress, modelTags, modelStep);
+        Recipe res = new Recipe(modelName);
+        //Setter methods/overloaded constructor
+        return res;
```

Comment:
SLAP and design can be *a lot* better

- Each model can have a `toDbSchema` method and inherit from a `JSONAdaptable` interface
- In the constructor for this, simply pass in the field objects natively and call the `toDbSchema` as opposed to overloading the `toString`
  - Can also have these fields be an `Optional` to set the JSON if present
    - RecipePortion
    - RecipeDuration
  
- Each model can also have a `fromDbSchema`